Upgrade to .NET 9 and OTAPI Static Hooks for ARM64

SignatureBeef opened the pull request:
### This PR aims for the following:
- Target frameworks & workflows updated to net9.0
- OTAPI and other packages updated
- Moved to [generated OTAPI static hooks](https://github.com/SignatureBeef/ModFramework.NET/blob/default/ModFramework/Emitters/HookEmitter.cs#L581), making MonoMod RD optional/supplemental (though, OTAPI.Runtime.dll is still included for those still utilising it, and works on net9)
  - _Note: File scoped namespaces was utilised, which is why the diff is more than expected._
- Expanded workflow to test each platform
- Reduced OTAPI/ModFw console outputs (e.g. no tile ctor print, short version codes etc)

### Considerations that may flow through to a TShock PR (or sooner if desired):
- OTAPI Static Hook namespace naming preferences e.g. `HookEvents.*` vs MonoMod RD `On.*`
- OTAPI Static Hook EventArgs property naming preferences e.g. `ContinueExecution`, `HookReturnValue`
- If any extra static hooks are needed, current list includes most methods from the following Terraria types: 
  - `Main, Item, NetMessage, Netplay, NPC, WorldGen, Chat.ChatHelper, IO.WorldFile, Net.NetManager, Projectile, RemoteClient, Liquid, Program.`
- Apple Silicon / ARM64 testing
- Plugin updates: at the moment should just be a matter of ticking up to net9.0, and matching dependencies/updating

### Screenshots

#### Windows x64
![image](https://github.com/user-attachments/assets/96a0b4ee-570e-424c-887b-248bc0556c22)
![image](https://github.com/user-attachments/assets/9bd52bdb-008b-428f-800a-299ab82a6c70)


#### MacOS Arm64
<img width="908" alt="image" src="https://github.com/user-attachments/assets/57076b79-a299-48e1-adca-62a6a31b7d43" />
<img width="923" alt="image" src="https://github.com/user-attachments/assets/efdd73e4-06d1-4926-a56e-18caffdd5059" />
Joining Mac Arm64 from Win64:

![image](https://github.com/user-attachments/assets/a0c8a18b-dc1c-469c-8ba2-f86116a5a634)

SignatureBeef committed:
Update to .NET 9

SignatureBeef committed:
Move to OTAPI static hooks

SignatureBeef committed:
Expand tests to multiple platforms

SignatureBeef committed:
Merge remote-tracking branch 'upstream/general-devel' into net9-upgrade

SignatureBeef committed:
Utilize nunit TestContext instead of console

Arthri reviewed (approved):
I have not tested the changes in game, however, the code changes look good. Thank you for working on this

SignatureBeef committed:
OTAPI update for MMRD 25.x + test to confirm

Excluding ARM64 as expected to currently fail / not supported

SignatureBeef committed:
Correct test variable initialisation

SignatureBeef committed:
Set Main.rand within server init test

Avoiding the use of DedServ.

SignatureBeef committed:
Amend EnsureInitialised init variable

SignatureBeef committed:
Merge branch 'Pryaxis:general-devel' into net9-upgrade

SignatureBeef committed:
Fix OnConnectionAccepted incorrectly allowing continuation

SignatureBeef committed:
Use OTAPI.Common.VersionShort for version printing

SignatureBeef committed:
OTAPI 3.2.4 - improve boss bag hook

CC:
https://github.com/SignatureBeef/Open-Terraria-API/pull/131
https://github.com/Pryaxis/TSAPI/issues/252
https://github.com/Pryaxis/TShock/issues/2998

QuiCM commented:
I too have not tested in game, but LGTM

hakusaro commented:
@SignatureBeef I merged https://github.com/Pryaxis/TSAPI/pull/253, can you rebase?

SignatureBeef committed:
Merge remote-tracking branch 'upstream/general-devel' into net9-upgrade

SignatureBeef committed:
Migrate UpdateConnectedClients to static hook
